Library/Homebrew/dev-cmd/bump-formula-pr.rb

```diff
@@ -298,27 +300,32 @@ def bump_formula_pr
         ohai "create pull request with GitHub API"
       else
 
-        begin
-          response = GitHub.create_fork(formula.tap.full_name)
-          # GitHub API responds immediately but fork takes a few seconds to be ready.
-          sleep 3
-
-          if system("git", "config", "--local", "--get-regexp", "remote\..*\.url", "git@github.com:.*")
-            remote_url = response.fetch("ssh_url")
-          else
-            remote_url = response.fetch("clone_url")
-          end
-          username = response.fetch("owner").fetch("login")
-        rescue GitHub::AuthenticationFailedError => e
-          raise unless e.github_message.match?(/forking is disabled/)
-
-          # If the repository is private, forking might be disabled.
-          # Create branches in the repository itself instead.
+        if args.no_fork?
           remote_url = Utils.popen_read("git remote get-url --push origin").chomp
           username = formula.tap.user
-        rescue *GitHub.api_errors => e
-          formula.path.atomic_write(backup_file) unless args.dry_run?
-          odie "Unable to fork: #{e.message}!"
+        else
+          begin
```

Comment:
Maybe this could all be extracted to another function to make use of e.g. early returns?